<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FilterDialog</class>
 <widget class="QDialog" name="FilterDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>New Filter</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0">
      <widget class="QLabel" name="labelMC">
       <property name="text">
        <string>MC: 1.X</string>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QComboBox" name="comboBoxCat">
       <property name="minimumSize">
        <size>
         <width>160</width>
         <height>0</height>
        </size>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::AdjustToMinimumContentsLengthWithIcon</enum>
       </property>
       <property name="iconSize">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
       <item>
        <property name="text">
         <string>Select category</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Algorithm helpers</string>
        </property>
        <property name="icon">
         <iconset resource="../icons.qrc">
          <normaloff>:/icons/helper.png</normaloff>:/icons/helper.png</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Quad-structure</string>
        </property>
        <property name="icon">
         <iconset resource="../icons.qrc">
          <normaloff>:/icons/quad.png</normaloff>:/icons/quad.png</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Structures</string>
        </property>
        <property name="icon">
         <iconset resource="../icons.qrc">
          <normaloff>:/icons/stronghold.png</normaloff>:/icons/stronghold.png</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Biome filter</string>
        </property>
        <property name="icon">
         <iconset resource="../icons.qrc">
          <normaloff>:/icons/overworld.png</normaloff>:/icons/overworld.png</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Nether biomes</string>
        </property>
        <property name="icon">
         <iconset resource="../icons.qrc">
          <normaloff>:/icons/nether.png</normaloff>:/icons/nether.png</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>End biomes</string>
        </property>
        <property name="icon">
         <iconset resource="../icons.qrc">
          <normaloff>:/icons/the_end.png</normaloff>:/icons/the_end.png</iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Others</string>
        </property>
        <property name="icon">
         <iconset resource="../icons.qrc">
          <normaloff>:/icons/slime.png</normaloff>:/icons/slime.png</iconset>
        </property>
       </item>
      </widget>
     </item>
     <item row="0" column="4">
      <widget class="QComboBox" name="comboBoxType">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimumSize">
        <size>
         <width>260</width>
         <height>0</height>
        </size>
       </property>
       <property name="maxVisibleItems">
        <number>20</number>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QComboBox::AdjustToMinimumContentsLengthWithIcon</enum>
       </property>
       <property name="iconSize">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
       <item>
        <property name="text">
         <string>Select filter</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="labelFilterType">
       <property name="text">
        <string>Filter type:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="Collapsible" name="collapseDescription" native="true"/>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBoxPosition">
         <property name="title">
          <string>Location</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="4" column="0">
           <widget class="QLabel" name="labelRelative">
            <property name="text">
             <string>Location is relative to:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QSpinBox" name="spinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>99</number>
            </property>
           </widget>
          </item>
          <item row="4" column="1" colspan="2">
           <widget class="QComboBox" name="comboBoxRelative">
            <item>
             <property name="text">
              <string>World origin</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineRadius">
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QCheckBox" name="buttonArea">
            <property name="toolTip">
             <string>Specify manual rectangle rather than square</string>
            </property>
            <property name="text">
             <string>Custom area</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelSpinBox">
            <property name="text">
             <string>Instances within area:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelSquareArea">
            <property name="text">
             <string>Within centred square of side:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="3">
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="labelX1">
              <property name="toolTip">
               <string>Lower bound (inclusive)</string>
              </property>
              <property name="text">
               <string>X&lt;sub&gt;1&lt;/sub&gt;:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEditX1">
              <property name="toolTip">
               <string>Lower bound (inclusive)</string>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="labelZ1">
              <property name="toolTip">
               <string>Lower bound (inclusive)</string>
              </property>
              <property name="text">
               <string>Z&lt;sub&gt;1&lt;/sub&gt;:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEditZ1">
              <property name="toolTip">
               <string>Lower bound (inclusive)</string>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="labelX2">
              <property name="toolTip">
               <string>Upper bound (inclusive)</string>
              </property>
              <property name="text">
               <string>X&lt;sub&gt;2&lt;/sub&gt;:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEditX2">
              <property name="toolTip">
               <string>Upper bound (inclusive)</string>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="labelZ2">
              <property name="toolTip">
               <string>Upper bound (inclusive)</string>
              </property>
              <property name="text">
               <string>Z&lt;sub&gt;2&lt;/sub&gt;:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEditZ2">
              <property name="toolTip">
               <string>Upper bound (inclusive)</string>
              </property>
              <property name="text">
               <string>0</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QTabWidget" name="tabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <property name="usesScrollButtons">
          <bool>false</bool>
         </property>
         <widget class="QWidget" name="tabBiomes">
          <attribute name="title">
           <string>Biomes</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="1" column="3">
            <widget class="QLineEdit" name="lineY">
             <property name="toolTip">
              <string>Vertical position in blocks</string>
             </property>
             <property name="text">
              <string>63</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QCheckBox" name="checkApprox">
             <property name="toolTip">
              <string>Enables optimizations that trade accuracy for speed</string>
             </property>
             <property name="text">
              <string>Approximate</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="labelY">
             <property name="text">
              <string>Sample at height (Y):</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="4">
            <widget class="QScrollArea" name="scrollArea">
             <property name="widgetResizable">
              <bool>true</bool>
             </property>
             <widget class="QWidget" name="scrollBiomes">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>568</width>
                <height>249</height>
               </rect>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <property name="sizeConstraint">
                <enum>QLayout::SetMinAndMaxSize</enum>
               </property>
               <item>
                <layout class="QGridLayout" name="gridLayoutBiomes"/>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
           <item row="2" column="0" colspan="4">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QPushButton" name="buttonUncheck">
               <property name="text">
                <string>uncheck all</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="buttonInclude">
               <property name="text">
                <string>include all</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="buttonExclude">
               <property name="text">
                <string>exclude all</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="1">
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabTemps">
          <attribute name="title">
           <string>Temperature Categories</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Select how many entries of each temperature category are required.</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayoutTemps"/>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabVariants">
          <attribute name="title">
           <string>Variant</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="0" column="1">
            <widget class="QCheckBox" name="checkAbandoned">
             <property name="text">
              <string>Abandoned</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QCheckBox" name="checkStartPiece">
             <property name="toolTip">
              <string>Allow only structures that start with one of the selected pieces</string>
             </property>
             <property name="text">
              <string>Filter starting piece</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QScrollArea" name="scrollVariants">
             <property name="widgetResizable">
              <bool>true</bool>
             </property>
             <widget class="QWidget" name="scrollAreaWidgetContents_2">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>568</width>
                <height>285</height>
               </rect>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <layout class="QGridLayout" name="gridLayoutVariants"/>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="2" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonCancel">
       <property name="text">
        <string>Cancel</string>
       </property>
       <property name="icon">
        <iconset theme="dialog-cancel">
         <normaloff/>
        </iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonOk">
       <property name="text">
        <string>Ok</string>
       </property>
       <property name="icon">
        <iconset theme="dialog-ok">
         <normaloff/>
        </iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Collapsible</class>
   <extends>QWidget</extends>
   <header>src/collapsible.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>buttonOk</tabstop>
  <tabstop>buttonCancel</tabstop>
  <tabstop>comboBoxCat</tabstop>
  <tabstop>comboBoxType</tabstop>
  <tabstop>spinBox</tabstop>
  <tabstop>lineRadius</tabstop>
  <tabstop>buttonArea</tabstop>
  <tabstop>lineEditX1</tabstop>
  <tabstop>lineEditZ1</tabstop>
  <tabstop>lineEditX2</tabstop>
  <tabstop>lineEditZ2</tabstop>
  <tabstop>comboBoxRelative</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>buttonUncheck</tabstop>
  <tabstop>buttonInclude</tabstop>
  <tabstop>buttonExclude</tabstop>
  <tabstop>scrollArea</tabstop>
 </tabstops>
 <resources>
  <include location="../icons.qrc"/>
 </resources>
 <connections/>
</ui>
